### DISCUSSION TITLE
Resnet50 .. res_r50_vd_db.yml Configuration Help.

### DISCUSSION THREAD

**Discussion by Ibrah-N** [PaddlePaddle/PaddleOCR#14994] [Q&A]
Hi, Everyone 
I hope you all are doing well. Actually I have been used the Res_r50_vd_db.yml  configuration for training the resnet50 model but doesn't giving me the good results because I have the images of 2048x2048 while I don't have complete information of how to setup the model configuration for these larger images to get the highest accuracy as possible.

So could you suggest some key parameters changing that let me achieve the highest accuracy for the larger images training of 2048x2048.

Thanks You.

**Comment by GreatV** [comment]
To train ResNet50 effectively on large images like 2048x2048 using the res_r50_vd_db.yml config (DB algorithm for text detection), you’ll need to make several crucial adjustments. The default configuration is optimized for smaller images (like 640x640), which explains why your performance is not optimal for high-res inputs.

Here are the key areas you should revise:

1. 💡 Data Preprocessing Adjustments

a. Crop Size in EastRandomCropData
Currently:
EastRandomCropData:
  size: [640, 640]

→ Change to better reflect your input:
EastRandomCropData:
  size: [1024, 1024]  or even up to [1536, 1536]

Reason: Larger crop sizes preserve more context and detail, especially for 2048x2048 images. Just make sure it fits in your GPU memory.

b. Resize Augmentation
The current IaaAugment has:
Resize size: [0.5, 3]

→ You might want to constrain this to a more stable range like [1.0, 2.0] especially if the images are already large, to avoid distortion.

2. 🧠 Model Input Size during Evaluation

Currently in DetResizeForTest:
image_shape: [736, 1280]

→ For large images, change this to something like:
image_shape: [1536, 2048]

or,
image_shape: [2048, 2048]

Just ensure it reflects your input aspect ratio and fits in memory.

3. 🧪 Training Strategy

a. Lower batch_size_per_card
Large images consume more GPU memory.

→ Reduce batch size to something like:
batch_size_per_card: 2 or 4

b. Adjust learning rate accordingly:
If batch size is decreased significantly, consider reducing initial learning rate (e.g., to 0.0005) or using a learning rate schedule.

4. 🧼 Data Augmentations

The Resize in IaaAugment along with random cropping might remove important details. You can try disabling random cropping temporarily or use random cropping with larger areas retained. Alternatively, you may consider other spatial augmentations more suitable for large images like RandomScale, RandomRotate, etc.

5. ✅ Eval Image Size and Accuracy

Make sure the test resolution approximates the input resolution. Otherwise, predicted results might be blurry or mismatched.

→ DetResizeForTest:
image_shape: [2048, 2048]

Note: You might want to try turning off resizing altogether if the model supports arbitrary size inputs.

6. 💾 Save performance

Set:
cal_metric_during_train: True
→ This allows monitoring of hmean during training for better checkpoints.

Optional:
- Switch optimizer to SGD for better generalization in large-scale setups (though Adam can still be okay for fast convergence).

7. 🔍 Consider Patch-Based Inference

For evaluation/inference on 2048x2048 images with limited memory, you may run patch-wise inference and aggregate results.

Summary of Key Changes:
| Parameter | Suggested Value |
|----------|-----------------|
| EastRandomCropData.size | [1024, 1024] or [1536, 1536] |
| batch_size_per_card | 2 or 4 |
| IaaAugment Resize | [1.0, 2.0] |
| DetResizeForTest.image_shape | [2048, 2048] |
| cal_metric_during_train | True |

Implement the above changes incrementally while monitoring hmean and loss values during training and evaluation phases. Adjust further based on GPU memory constraints and overfitting observations.

Let me know if you'd like help with modifying the full .yml file accordingly.

---
*Response generated by 🤖 [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*

 ⚠️ This is an AI-generated response. Please verify any critical information.
